include/stdlib_coroutine.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+/**
+ * @file stdlib_coroutine.hpp
+ * @author Alecto Perez (perez.cs@pm.me)
+ * @brief This file defines a compatibility layer for C++20 coroutines
+ * @date 2021-09-23
+ *
+ * @copyright Copyright (c) 2021
+ *
+ */
+
+#if __has_include(<coroutine>)
+
+#if defined(__clang__) && !defined(__cpp_impl_coroutine)
+// This needs to be defined to enable the <coroutine> header in libc++ when
+// using clang
+#define __cpp_impl_coroutine 201902L
+#endif
+#include <coroutine>
+
+#ifdef __clang__
+
+// std::coroutine_handle and std::coroutine_traits need to be included in
+// std::experimental when using libc++ and clang together, however the others
```

Comment:
I am confused. This code seems trying to support clang + libstdc++, but the comment reads *when using libc++ and clang together*.

Reply:
The code is intended to support:

- Clang with the Clang implementation of the C++ Standard Library (libstdc++)
- Clang with the GCC implementation of the C++ Standard Library (libc++)
- GCC with the GCC implementation of the C++ Standard Library (libc++)

In all cases you should be able to use std:: for coroutines instead of having to use std::experimental!